src/Compilers/CSharp/Portable/Symbols/Synthesized/GeneratedNameParser.cs

```diff
@@ -170,5 +171,22 @@ internal static bool TryParseAnonymousTypeParameterName(string typeParameterName
             propertyName = null;
             return false;
         }
+
+        // <ContainingFile>FN__ClassName`OptionalArity
+        private static readonly Regex s_fileTypeOrdinalPattern = new Regex(@">F(\d)+__", RegexOptions.Compiled);
+
+        internal static bool TryParseFileTypeName(string unmangledTypeName, out int ordinal, out string originalTypeName)
```

Comment:
>unmangledTypeName [](http://example.com/codeflow?start=57&length=17)

Isn't this the "mangledTypeName" rather than "unmangled"?

Reply:
It's confusing but we're specifically reserving the term "mangling" to refer to CLS-defined forms of mangling, i.e. the arity suffix. That's why I used the term DecodeFileType at one point here. Perhaps Parse would be appropriate. That seems to be how we describe extracting info out of other generated names in here.